libraries/tools/ts_v2_utils.py

```diff
@@ -90,17 +276,42 @@ def index_tn_rc(lid, temp_dir, rc_dir, reporter=None):
             firstvs = None
             note_hashes = []
             for verse in verses:
-                if verse in ['.', '..', 'intro.md']:
+                if verse in ['.', '..', 'intro.md', '.DS_Store']:
                     continue
 
                 # notes = []
                 verse_file = os.path.join(chapter_dir, verse)
                 verse = verse.split('.')[0]
-                verse_body = read_file(verse_file)
+                try:
+                    verse_body = read_file(verse_file)
+                except Exception as e:
+                    if reporter:
+                        reporter.report_error('Failed to read file {}'.format(verse_file))
+                    raise e
 
                 verse_body = convert_rc_links(verse_body)
                 general_notes = note_general_re.search(verse_body)
 
+                # zero pad chapter to match chunking scheme
+                padded_chapter = chapter
+                while len(padded_chapter) < 3 and padded_chapter not in chunk_json:
+                    padded_chapter = padded_chapter.zfill(len(padded_chapter) + 1)
+                    # keep padding if match is found
+                    if padded_chapter in chunk_json:
+                        chapter = padded_chapter
+
+                # validate chapters
+                if pid != 'obs' and chapter not in chunk_json:
+                    raise Exception('Missing chapter "{}" key in chunk json while reading chunks for {}. RC: {}'.format(chapter, pid, rc_dir))
+
+                # zero pad verse to match chunking scheme
+                padded_verse = verse
+                while len(padded_verse) < 3 and chapter in chunk_json and padded_verse not in chunk_json[chapter]:
+                    padded_verse = padded_verse.zfill(len(padded_verse) + 1)
+                    # keep padding if match is found
+                    if padded_verse in chunk_json[chapter]:
+                        verse = padded_verse
```

Comment:
This is too complex. The `while` tests `chapter`, even though it's invariant throughout the loop.
Instead, consider using your `pad_to_match` function above.

Reply:
Done